src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -235,18 +237,24 @@ private void AddComponentStats(StringBuilder builder)
                 string peerHeights = $"(r/s):{(chainHeadersBehavior.BestReceivedTip != null ? chainHeadersBehavior.BestReceivedTip.Height.ToString() : peer.PeerVersion?.StartHeight + "*" ?? "-")}";
                 peerHeights += $"/{(chainHeadersBehavior.BestSentHeader != null ? chainHeadersBehavior.BestSentHeader.Height.ToString() : peer.PeerVersion?.StartHeight + "*" ?? "-")}";
 
+                // TODO: Need a snapshot cache so that not only currently connected peers are summed
+                string peerTraffic = $"R/S MB: {(peer.Counter.ReadBytes / 1048576.0):0.##}/{(peer.Counter.WrittenBytes / 1048576.0):0.##}";
+                totalRead += peer.Counter.ReadBytes;
+                totalWritten += peer.Counter.WrittenBytes;
+
                 string agent = peer.PeerVersion != null ? peer.PeerVersion.UserAgent : "[Unknown]";
                 peerBuilder.AppendLine(
                     "Peer:" + (peer.RemoteSocketEndpoint + ", ").PadRight(LoggingConfiguration.ColumnLength + 15) +
                     (" connected:" + (peer.Inbound ? "inbound" : "outbound") + ",").PadRight(LoggingConfiguration.ColumnLength + 7)
                     + peerHeights.PadRight(LoggingConfiguration.ColumnLength + 7)
+                    + peerTraffic.PadRight(LoggingConfiguration.ColumnLength + 7)
                     + " agent:" + agent);
             }
 
             int inbound = this.ConnectedPeers.Count(x => x.Inbound);
 
             builder.AppendLine();
-            builder.AppendLine($"======Connection====== agent {this.Parameters.UserAgent} [in:{inbound} out:{this.ConnectedPeers.Count() - inbound}]");
+            builder.AppendLine($"======Connection====== agent {this.Parameters.UserAgent} [in:{inbound} out:{this.ConnectedPeers.Count() - inbound}] [recv: {totalRead} sent: {totalWritten}]");
```

Comment:
I'd also add avg traffic per minute
That will require keeping connection time of a peer (btw connection uptime is also smth we can display)

Reply:
yeah good idea.